File: java/mission-to-mars-solid/src/main/java/mars/MouvementOuest2D.java
package mars;

public class MouvementOuest2D implements Mouvement2D {

	public Position avancer(Position position) {
		return new Position(position.getX() - 1, position.getY(), position.getZ());
	}
	
	public Position reculer(Position position) {
		return new Position(position.getX() + 1, position.getY(), position.getZ());
	}
}

File: python/mission-to-mars-solid/src/mars/MouvementOuest2D.py
from src.mars.Mouvement2D import Mouvement2D
from src.mars.Position import Position


class MouvementOuest2D(Mouvement2D):

    def avancer(self, position: Position) -> Position:
        return Position(position.x - 1, position.y, position.z)

    def reculer(self, position: Position) -> Position:
        return Position(position.x + 1, position.y, position.z)
